src/libcore/mem/manually_drop.rs

```diff
@@ -118,7 +118,8 @@ impl<T: ?Sized> ManuallyDrop<T> {
     ///
     /// This function runs the destructor of the contained value and thus the wrapped value
     /// now represents uninitialized data. It is up to the user of this method to ensure the
-    /// uninitialized data is not actually used.
+    /// uninitialized data is not actually used, and that this function is called at most once
+    /// for a given instance of `ManuallyDrop<T>`.
```

Comment:
I think clarifying this is a good call. However, by chaining it with ", and" you make it sound as if this was a new guarantee, whereas really this is covered by the first clause and we just state it for clarify.

So maybe end the first sentence there, and that a new one something like "In particular, this means that this function must be called at most once [...]"?